app/controllers/concerns/next_link_header.rb

```diff
@@ -0,0 +1,18 @@
+module NextLinkHeader
+  extend ActiveSupport::Concern
+
+private
+
+  def next_link_header(from_key_string, last_model, changed_since, per_page)
+    response.headers['Link'] = if last_model
+                                 header_content(from_key_string, (last_model.updated_at + 1.second).utc.iso8601, last_model.id, per_page)
```

Comment:
whats the purpose of adding one second?

Reply:
The idea is that we generate a new `changed_since` timestamp that is one second after the last object that is being returned as this page.

Having said that, I think that there is a bug here. I say we evaluate this PR independent of this aspect and address it later.

Reply:
What is the bug?

Reply:
If the last object in our list is in the middle of a result set, i.e. there is another page of results, then it shouldn't be the time of update of the last object on the current page + 1 second, it should be the time of update of the next object on the next page. I think this is hard to explain in words, this is part of what Tim was alluding to with his comment about the design, but I think it can be simply illustrated.

Reply:
Let me see if I understand correctly:

```
a { changed_since: 13:00:00 }
b { changed_since: 13:00:04 }
...
f { changed_since: 13:03:23 }
<page-break>
g { changed_since: 13:05:22 }
...
```

In our current implementation, if we return a page containing `a, b, ..., f`, the `changed_since` will be `f.changed_since + 1.second`, so `13:03:24`.

But you're saying the next item is at `13:05:22`, so that's what we should return. Why? It shouldn't change the functionality; all the items are sorted by their `changed_since` so querying from `13:03:24` and `13:05:22` should always return the same set of items.

Reply:
Yes.

You say "VERY tiny bug", but in fact one of the things I thought we should protect for is a batch publishing of providers that all happen within the same second ... I'm not sure how often that'll happen but it doesn't sound too outlandish to me.

Reply:
Correct. :)

Let's fix it. 👍

Reply:
We've added functionality to the `changed_since` scope in order to address this issue, and we should backport this fix to Providers.